metricplugin/api.go

```diff
@@ -0,0 +1,72 @@
+package metricplugin
+
+import (
+	"time"
+
+	bsmsg "github.com/ipfs/go-bitswap/message"
+	peer "github.com/libp2p/go-libp2p-core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+// A BitswapMessage is the type pushed to remote clients for recorded incoming
+// Bitswap messages.
+type BitswapMessage struct {
+	// The entries requested.
+	WantlistEntries []bsmsg.Entry `json:"wantlist_entries"`
+
+	// Whether this was a full wantlist.
+	FullWantList bool `json:"full_wantlist"`
+}
+
+// ConnectionEventType specifies the type of connection event.
+type ConnectionEventType int
+
+const (
```

Comment:
More of a question than a review: shouldn't these be realized through the [iota](https://golang.org/ref/spec#Iota) idiom?

Reply:
That's only nice for Go-Go interoperability and usually confuses people who don't know Go but need to understand the API :)